examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavigationFragment.kt

```diff
@@ -160,6 +163,17 @@ class BasicNavigationFragment : Fragment(), OnMapReadyCallback, FeedbackBottomSh
     override fun onSaveInstanceState(outState: Bundle) {
         super.onSaveInstanceState(outState)
         mapView.onSaveInstanceState(outState)
+        // This is not the most efficient way to preserve the route on a device rotation.
+        // This is here to demonstrate that this event needs to be handled in order to
+        // redraw the route line after a rotation.
+        directionRoute?.let {
+            outState.putString(PRIMARY_ROUTE_BUNDLE_KEY, it.toJson())
```

Comment:
does it work for huge routes more than `1 Mb` String? If no seems that we should think about different way how to restore route(`ViewModel` is the simplest way)

Reply:
There's a comment that points out it's not an efficient way to preserve the route. The purpose of the activity is to present a basic example for using the SDK. I think the more code we put in the examples the harder they are for a new developer to understand.

Reply:
I see. Cannot fully agree because samples have to be crash free, but I'm ok, it's a tricky point here